tests/datasource/fluent/test_snowflake_datasource.py

```diff
@@ -9,62 +11,98 @@
 from great_expectations.compatibility import pydantic
 from great_expectations.compatibility.snowflake import snowflake
 from great_expectations.data_context import AbstractDataContext
+from great_expectations.datasource.fluent import GxContextWarning
 from great_expectations.datasource.fluent.config_str import ConfigStr
 from great_expectations.datasource.fluent.snowflake_datasource import (
     SnowflakeDatasource,
     SnowflakeDsn,
 )
 from great_expectations.execution_engine import SqlAlchemyExecutionEngine
 
+if TYPE_CHECKING:
+    from pytest.mark.structures import ParameterSet
+
+VALID_DS_CONFIG_PARAMS: Final[Sequence[ParameterSet]] = [
+    param(
+        {
+            "connection_string": "snowflake://my_user:password@my_account/d_public/s_public?numpy=True"
+        },
+        id="connection_string str",
+    ),
+    param(
+        {"connection_string": "${MY_CONN_STR_PARTIAL}@${MY_PATH}?${MY_QUERY_PARAMS}"},
+        id="connection_string ConfigStr with required query params",
+    ),
+    param(
+        {"connection_string": "${MY_CONN_STR_FULL}"},
+        id="connection_string ConfigStr - required params part of sub",
+    ),
+    param(
+        {"connection_string": "${MY_CONN_STR_MIN}?${MY_QUERY_PARAMS}"},
+        id="connection_string ConfigStr - dedicated query params sub",
+    ),
```

Comment:
We want to disallow these, right? else we give up the ability to say "hey if you edit that db name, these are all the configured assets that will no longer work"

Reply:
Yeah, I think the best answer here would be to prevent config substitution in the `path` portion of the connection string.

But I think we should enforce that in the Cloud Backend rather than OSS. I am not 100% convinced of that yet, but that's my initial reaction.

Reply:
If we enforce it in the Cloud Backend and not here, we're creating space for migration pain for users who start using the OSS library and want to convert to a cloud account. Another way of asking this question: why is cloud migration pain an acceptable cost to pay to allow config substitution in connection string paths?

Reply:
We'll enforce this in OSS with a followup.